src/OnboardingSPA/pages/Steps/GetStarted/SiteTypeSetup/PrimarySite/index.js

```diff
@@ -37,130 +37,138 @@ const StepPrimarySetup = () => {
 		setIsDrawerSuppressed( true );
 		setDrawerActiveView( VIEW_NAV_GET_STARTED );
 		setIsHeaderNavigationEnabled( true );
+		getSiteClassificationsData();
 	}, [] );
 
-	const [ clickedIndex, changeCategory ] = useState( -1 );
-	const [ inputCategVal, changeInputCateg ] = useState( '' );
+	const [ siteClassData, setSiteClassData ] = useState();
+	const [ primaryCategory, changePrimaryCategory ] = useState( '' );
+	const [ inputFieldValue, setInputFieldValue ] = useState( '' );
 
-	const categoriesArray = content.categories;
-	const selectedPrimaryCategoryInStore = currentData?.data?.siteType?.primary;
+	/**
+	 * Function which fetches the Site Classifications
+	 *
+	 */
+	const getSiteClassificationsData = async () => {
+		let siteClassificationsData = await getSiteClassifications();
+		if ( siteClassificationsData?.body.length === 0 ) {
+			siteClassificationsData = fallbackDataTypes;
+		}
 
-	/**This condition fills the data in input box if the saved category isn't a subcategory from the content*/
-	if ( selectedPrimaryCategoryInStore && ! inputCategVal ) {
-		const found = categoriesArray.find(
-			( e ) => e.name === selectedPrimaryCategoryInStore
-		);
-		if ( ! found && selectedPrimaryCategoryInStore !== 'primaryCategory' )
-			changeInputCateg( selectedPrimaryCategoryInStore );
-	}
+		setSiteClassData( siteClassificationsData?.body );
+		changePrimaryCategory( currentData?.data?.siteType?.primary ?? '' );
+
+		if ( currentData?.data?.siteType?.labelPri === 'custom' ) {
```

Comment:
```suggestion
		if ( currentData?.data?.siteType?.labelPri === 'custom' ) {
```
Shouldn't this be `other` based on data required for analytics?

Reply:
The value in labelPri won't be used rather the primary value would be used for analytics I suppose, this is just a way to determine if the user typed the data or not as other is itself a category in primary types, we can even further more determine if the data was typed or other was selected as the data typed would be stored too, thus extracting more data.

Reply:
Let's switch when we make that data structure change.

Reply:
Let's switch when we make that data structure change.